### ISSUE TITLE
Codebuild in deploy stage adds inline policy to adf-codebuild-role, exceeds policy size limit

### ISSUE THREAD

**Issue by eriklz** [awslabs/aws-deployment-framework#217]
When configuring a pipeline that uses CodeBuild in the deploy stage, an inline policy is added to adf-codebuild-role. With enough pipelines doing that, the 10KB size limit for inline policies in a role is exceeded, and additional pipelines cannot be added. See also attached picture.

![image](https://user-images.githubusercontent.com/45368867/72797733-f2aa2880-3c41-11ea-9981-e943589fdd73.png)

It seems that the CDK, in the Project class for aws_codebuild, the constructor adds policies to the provided role, which may perhaps be the reason this inline policy gets added. The inline policy includes logging permissions as well as permissions for KMS key (which is already in adf-codebuild-policy).

![image](https://user-images.githubusercontent.com/45368867/72798317-1ae65700-3c43-11ea-9487-356634f3fc76.png)

I updated /adf-bootstrap/deployment/global.yml and adf-codebuild-policy in there to include adf-deploy-* log groups and not only adf-build-* log groups. However, that did not have any effect on the inline policies, it was still generated as before.

```yaml
          - Effect: Allow
            Action:
              - "logs:CreateLogGroup"
              - "logs:CreateLogStream"
              - "logs:PutLogEvents"
            Resource:
              - !Sub arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/codebuild/adf-build-*
              - !Sub arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/codebuild/adf-deploy-*
```

Bottom line is that it is not possible to add additional pipelines after a while, when the default adf-codebuild-role is used.

**Comment by bundyfx** [comment]
Thanks @eriklz - Taking a look into this now, it appears as you described, hitting the [10,240 char limit in a policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-limits.html) - Since its duplicates however we should be able to sort this one pretty easily.

**Comment by eriklz** [comment]
I modified the code in /adf-build/shared/cdk/cdk_constructs/adf_codebuild.py to set mutable=False in the Role.from_role_arn() calls in case _build_role equals ADF_DEFAULT_BUILD_ROLE.
Together with the addition of log-group:/aws/codebuild/adf-deploy-* in global.yml this looks like it works so far. Only adf-codebuild-policy is present as policy in adf-codebuild-role after rebuilding pipelines.
